### PR TITLE
Initial porting guide

### PR DESCRIPTION
**Pull request by utzig**
I fixed the review issues, so my suggestion is to merge as is and open new PR's if there's more stuff that needs to be fixed.

### CODE REVIEW

**Comment by utzig** [comment]
@kYc0o Suggestions are welcome!

File: doc/PORTING.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+an initialized `boot_rsp` which has pointers to the location of the image
+where the target firmware is located which can be used to jump to.
+
+## Flash access and flash Map
+
+* Regarding flash access the bootloader has two requirements:
+
+### hal_flash_align
+
+`mcuboot` needs to know the alignment of the flash. To get this information it
```
Comment (d3zd3z) [inline doc/PORTING.md]: `mcuboot` needs to know the minimum write size (and alignment) of the flash.

File: doc/PORTING.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+calls `hal_flash_align`.
+
+TODO: this needs to die and move to flash_map...
+
+### flash_map
+
+The bootloader requires a `flash_map` to be able to know how the flash is
```
Comment (d3zd3z) [inline doc/PORTING.md]: Given that partition is the ordinary term for this, and map is kind of an unusual term, perhaps partition shouldn't be in quotes, and maybe map should be. Maybe: "The bootloader needs to now how the flash is partitioned, using a `flash_map`..."

**Comment by utzig** [comment]
@d3zd3z
> Perhaps we should create a ticket to track this, and possible mention this document needing to be updated.

You mean track possible use of `tinycrypt` without requiring `mbed`?

File: doc/PORTING.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+`mcuboot` to a new target `OS`.
+
+# Requirements
+
+* `mcuboot` requires that the target provides a `flash` API with ability to
+  get flash alignment and read/write/erase individual sectors.
+
+* `mcuboot` doesn't bundle a cryptographic library, which means the target
+  OS must already have it bundled. The supported libraries at the moment are
+  `tinycrypt` and `mbed TLS`.
```
Comment (utzig) [inline doc/PORTING.md]: You mean track possible use of `tinycrypt` without requiring `mbed TLS`?

File: doc/PORTING.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
+# Porting How-To
+
+This document describes the requirements and necessary steps required to port
+`mcuboot` to a new target `OS`.
+
+# Requirements
+
+* `mcuboot` requires that the target provides a `flash` API with ability to
+  get flash alignment and read/write/erase individual sectors.
```
Comment (utzig) [inline doc/PORTING.md]: "write size" sounds good to me.

File: doc/PORTING.md (hunk 2 above)
Comment (utzig) [inline doc/PORTING.md]: I removed the quotes around partition. Still thinking about a better way to use the terminology here.

**Pull request merged**